src/rpc/blockchain.cpp

```diff
@@ -2329,30 +2360,130 @@ UniValue CreateUTXOSnapshot(
 
-    SnapshotMetadata metadata{tip->GetBlockHash(), stats.coins_count, tip->nChainTx};
+        afile << metadata;
 
-    afile << metadata;
+        COutPoint key;
+        Coin coin;
+        unsigned int iter{0};
+
+        while (pcursor->Valid()) {
+            if (iter % 5000 == 0) node.rpc_interruption_point();
+            ++iter;
+            if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
+                afile << key;
+                afile << coin;
+            }
 
-    COutPoint key;
-    Coin coin;
-    unsigned int iter{0};
-
-    while (pcursor->Valid()) {
-        if (iter % 5000 == 0) node.rpc_interruption_point();
-        ++iter;
-        if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
-            afile << key;
-            afile << coin;
+            pcursor->Next();
         }
 
-        pcursor->Next();
+        afile.fclose();
+        break;
     }
+    case UTXOSnapshotFormat::SQLITE: {
+#ifdef USE_SQLITE
+        LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to sqlite DB file %s",
+                                   tip->nHeight, tip->GetBlockHash().ToString(), fs::PathToString(path)));
+
+        sqlite3* db;
+        int ret;
+        std::string path_string = fs::PathToString(path);
+
+        if (!ContainsNoNUL(path_string)) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid path");
+        }
+
+        ret = sqlite3_open(path_string.c_str(), &db);
+        if (ret != SQLITE_OK) {
+            sqlite3_close(db);
+            throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("Unable to open database file %s", path.u8string()));
+        }
+
+        // We have no need for rollbacks here, so we can set the `journal_mode` to 'off'. See: https://www.sqlite.org/pragma.html#pragma_journal_mode
+        ret = sqlite3_exec(db, "PRAGMA journal_mode=off;", nullptr, nullptr, nullptr);
+        if (ret != SQLITE_OK) {
+            throw std::runtime_error(strprintf("SQLiteDatabase: Failed to set 'journal_mode=off': %s\n", sqlite3_errstr(ret)));
+        }
 
-    afile.fclose();
+        // Create and populate 'metadata' table
+        ret = sqlite3_exec(db, "CREATE TABLE metadata(blockhash TEXT, blockheight INT);", nullptr, nullptr, nullptr);
+        if (ret != SQLITE_OK) {
+            sqlite3_close(db);
+            throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("SQLiteDatabase: Failed to create 'metadata' table: %s\n", sqlite3_errstr(ret)));
+        }
+        ret = sqlite3_exec(db, strprintf("INSERT INTO metadata VALUES ('%s', %d)", tip->GetBlockHash().ToString(), tip->nHeight).c_str(), nullptr, nullptr, nullptr);
+        if (ret != SQLITE_OK) {
+            sqlite3_close(db);
+            throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("SQLiteDatabase: Failed to insert into 'metadata' table: %s\n", sqlite3_errstr(ret)));
+        }
+
+        const char* sql = "CREATE TABLE utxos(txid TEXT, vout INT, value INT, coinbase INT, height INT, scriptpubkey TEXT);";
```

Comment:
`vout` is the wrong term

Reply:
What would be a better alternative? Simply `n`? Right now `vout` is already used in quite some RPC calls to refer to an outpoint's index (e.g. in the resut of `scantxoutset`).